### ISSUE TITLE
Loading the trained model predicts different results than using test.py

### ISSUE THREAD

**Issue by DejaVuyan** [junyanz/pytorch-CycleGAN-and-pix2pix#1339]
I trained a modified Pix2PIx model，and I want to load the model by my own data,such as:
```
import torch
from models.networks import LocalEnhancer
from data.base_dataset import BaseDataset, get_params, get_transform
from PIL import Image
from torchvision import transforms
import functools
import torch.nn as nn
from models.base_model import BaseModel

img_path = r'/data/yuzun/MRI_data/reged_ADNI/ADNI_reged_AB_img/test/168_S_6049_2017_0.png'
save_path = r'/data/yuzun/ours_model/L1_1/latest_net_G.pth'

n_downsample_global = 3
n_blocks_global = 9
n_local_enhancers = 1
n_blocks_local = 3
input_nc = 1
output_nc = 1
ngf = 64
norm_layer = functools.partial(nn.BatchNorm2d, affine=True, track_running_stats=True)

net = LocalEnhancer(input_nc, output_nc, ngf, n_downsample_global, n_blocks_global,
                    n_local_enhancers, n_blocks_local,norm_layer)
basemodel = BaseModel()

state_dict = torch.load(save_path, map_location=str("cuda:0"))
if hasattr(state_dict, '_metadata'):
    del state_dict._metadata

# patch InstanceNorm checkpoints prior to 0.4
for key in list(state_dict.keys()):  # need to copy keys here because we mutate in loop
    basemodel.__patch_instance_norm_state_dict(state_dict, net, key.split('.'))
net.load_state_dict(state_dict)

net.load_state_dict( torch.load(save_path) )
#net.eval()

# load image
trans = transforms.Compose([
                            #transforms.Grayscale(1),
                            transforms.Resize([112,112]),
                            transforms.ToTensor(),
                            transforms.Normalize((0.5,), (0.5,))
                            ])

A = Image.open(img_path).convert('L')
A = trans(A).unsqueeze(dim=0)
output = net(A)
toPIL = transforms.ToPILImage()
#togray = transforms.Grayscale(1)
#img = togray( toPIL(output.squeeze(dim=0)) )

img = toPIL(output.squeeze(dim=0))
img.save('test3.png')
```
But the model predicted worse results than I predicted using `test.py`,why is that?

**Comment by DejaVuyan** [comment]
I have found the proble, when I save image by `transforms.ToPILImage()`,I have not converted the image from normalization to original `[0,255],unit 8` type;So my picture doesn't look like a normal picture.Can someone tell me how to use `transforms.ToPILImage()` correctly to save pictures？

**Comment by junyanz** [comment]
We have a function called [tensor2im](https://github.com/junyanz/pytorchtensor2im), which might be helpful.
